velox/exec/MergeJoin.cpp

```diff
@@ -247,7 +247,9 @@ int32_t MergeJoin::compare(
     if (!compare.has_value()) {
       // The SQL semantics of Presto and Spark will always return false if
       // comparing a NULL value with any other value.
-      return -1;
+      // If right side is null, and left side is not null, should return 1
+      // So the following process will catch up rightInput_ with input_
+      return batch->childAt(keys[i])->isNullAt(index) ? -1 : 1;
```

Comment:
Could we consider changing the null handle mode to kNullAsValue [here](https://github.com/facebookincubator/velox/blob/d8a7a5e0984069785085d607920e20345b8cb3fe/velox/exec/MergeJoin.cpp#L239)? It appears that the `compareNulls  `method already handles this case appropriately: https://github.com/facebookincubator/velox/blob/d8a7a5e0984069785085d607920e20345b8cb3fe/velox/vector/BaseVector.h#L910-L928